flink-runtime/src/test/java/org/apache/flink/runtime/io/disk/FileChannelManagerImplTest.java

```diff
@@ -0,0 +1,187 @@
+
+		try {
+			fileChannelManagerTestProcess.startProcess();
+
+			// Waits till netty shuffle environment has created the tmp directories.
+			Deadline deadline = Deadline.now().plus(TEST_TIMEOUT);
+
+			while (!fileOrDirExists(signalDir, COULD_KILL_SIGNAL_FILE) && deadline.hasTimeLeft()) {
+				Thread.sleep(100);
+			}
+
+			if (!fileOrDirExists(signalDir, COULD_KILL_SIGNAL_FILE) ||
+					!fileOrDirExists(fileChannelDir, DIR_NAME_PREFIX)) {
+				fail("The file channel manager test process does not create target directories in time, " +
+						"its output is: \n" + fileChannelManagerTestProcess.getProcessOutput());
+			}
+
+			Process kill = Runtime.getRuntime().exec("kill " + fileChannelManagerTestProcess.getProcessId());
+			kill.waitFor();
+			assertEquals("Failed to send SIG_TERM to process", 0, kill.exitValue());
+
+			deadline = Deadline.now().plus(TEST_TIMEOUT);
+			while (fileChannelManagerTestProcess.isAlive() && deadline.hasTimeLeft()) {
+				Thread.sleep(100);
+			}
+
+			if (fileChannelManagerTestProcess.isAlive()) {
+				fail("The file channel manager test process does not terminate in time, its output is: \n" + fileChannelManagerTestProcess.getProcessOutput());
+			}
+
+			// Checks if the directories are cleared.
+			assertThat("The file channel manager test process does not remove the tmp shuffle directories after termination, " +
+							"its output is \n" + fileChannelManagerTestProcess.getProcessOutput(),
+					fileOrDirExists(fileChannelDir, DIR_NAME_PREFIX),
+					is(false));
+		} finally {
+			fileChannelManagerTestProcess.destroy();
+		}
+	}
+
+	private boolean fileOrDirExists(File rootTmpDir, String namePattern) {
+		File[] candidates = rootTmpDir.listFiles((dir, name) -> name.contains(namePattern));
+		return candidates != null && candidates.length > 0;
+	}
+
+	/**
+	 * The {@link FileChannelManagerCleanupRunner} instance running in a separate JVM process.
+	 */
+	private static class FileChannelManagerTestProcess extends TestJvmProcess {
+		private final boolean callerHasHook;
+		private final String tmpDirectories;
+		private final String couldKillSignalFilePath;
+
+		FileChannelManagerTestProcess(boolean callerHasHook, String tmpDirectories, String couldKillSignalFilePath) throws Exception {
+			this.callerHasHook = callerHasHook;
+			this.tmpDirectories = tmpDirectories;
+			this.couldKillSignalFilePath = couldKillSignalFilePath;
+		}
+
+		@Override
+		public String getName() {
+			return "File Channel Manager Test";
+		}
+
+		@Override
+		public String[] getJvmArgs() {
+			return new String[]{
+					Boolean.toString(callerHasHook),
+					tmpDirectories,
+					couldKillSignalFilePath
+			};
+		}
+
+		@Override
+		public String getEntryPointClassName() {
+			return FileChannelManagerCleanupRunner.class.getName();
+		}
+	}
+
+	/**
+	 * The entry point class to test the file channel manager cleanup with shutdown hook.
+	 */
+	public static class FileChannelManagerCleanupRunner {
```

Comment:
public -> private

Reply:
I am a little worry since that the class is used as an entry point class of one process. Although it seems  not cause error with the current JDK, I am afraid it may cause unstableness in the future.